### PR TITLE
QPPCT-693: Error message for invalid performanceRateUuid is incorrect

### PR DESCRIPTION
**Pull request by halprin**
### Information
- JIRA story QPPCT-693.

### Changes proposed in this PR:
- Before, when the performance UUID was incorrect, we were generating an error message that stated the performance rate UUID needed to be what was already _provided_ in the QRDA3 file versus the _expected_ UUIDs.
- We now supply the expected UUIDs for the performance rate.

### Checklist 
- [x] All JUnit tests pass (`mvn clean verify`).
- [x] New unit tests written to cover new functionality.
- [x] Added and updated JavaDocs for non-test classes and methods.
- [x] No local design debt. Do you feel that something is "ugly" after your changes?
- [x] Updated documentation (`README.md`, etc.) depending if the changes require it.

### CODE REVIEW

File: test-commons/src/main/java/gov/cms/qpp/test/helper/AwsTestHelper.java [hunk 1]
```diff
@@ -19,11 +21,25 @@
  * A helper class for aws testing
  */
 public class AwsTestHelper {
-	private static AmazonDynamoDB dynamoClient = AmazonDynamoDBClientBuilder.standard().build();
-	private static AmazonS3 s3Client = AmazonS3ClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
 	public static final String TEST_DYNAMO_TABLE_NAME = "qpp-qrda3converter-acceptance-test";
+
+	private static AmazonDynamoDB dynamoClient = AmazonDynamoDBClientBuilder.standard().withRegion(Regions.US_EAST_1).build();
```
Comment (omnivion) [inline test-commons/src/main/java/gov/cms/qpp/test/helper/AwsTestHelper.java]: why not also final

File: converter/src/test/java/gov/cms/qpp/acceptance/QualityMeasureIdRoundTripTest.java [hunk 2]
```diff
@@ -58,16 +58,17 @@ void testMeasureCMS68v6PerformanceRateUuid() throws IOException {
 		}
 
 		String measureId = "CMS68v6";
-		String incorrectId = "00000000-0000-0000-0000-1NV4L1D";
+		String correctId = MeasureConfigs.getConfigurationMap()
+			.get("40280381-52fc-3a32-0153-3d64af97147b").getSubPopulation().get(0).getNumeratorUuid();
```
Comment (saquino0827) [inline converter/src/test/java/gov/cms/qpp/acceptance/QualityMeasureIdRoundTripTest.java]: But 1NV4L1D is so 1337

File: test-commons/src/main/java/gov/cms/qpp/test/helper/AwsTestHelper.java (hunk 1 above)
Comment (saquino0827) [reply test-commons/src/main/java/gov/cms/qpp/test/helper/AwsTestHelper.java]: I believe this was changed in another PR to be final. Probably just a matter of merging.

**Comment by sfradkin** [comment]
@halprin: Unless I missed it, is there a test to verify that the new code will aggregate multiple numerator UUIDs for a measure that supports that?

**Comment by halprin** [comment]
@sfradkin Oops, nope.  I'll add a test.

**Comment by halprin** [comment]
And added.

File: converter/src/main/java/gov/cms/qpp/conversion/validate/MipsQualityMeasureIdValidator.java [hunk 3]
```diff
@@ -100,7 +101,10 @@ private void validatePerformanceRateUuid(Node node, List<SubPopulation> subPopul
 					.orElse(null);
 
 			if (subPopulation == null) {
-				addPerformanceRateValidationMessage(node, performanceUuid);
+				String expectedPerformanceUuids = StringHelper.join(subPopulations.stream()
+					.map(SubPopulation::getNumeratorUuid)
+					.collect(Collectors.toSet()), ", ", "or ");
```
Comment (clydet) [inline converter/src/main/java/gov/cms/qpp/conversion/validate/MipsQualityMeasureIdValidator.java]: for readability, mind pulling this out of the join call?

File: converter/src/test/java/gov/cms/qpp/conversion/validate/QualityMeasureIdValidatorTest.java [hunk 4]
```diff
@@ -349,12 +352,32 @@ void testPerformanceRateUuidFail() {
...
+	@Test
+	void testPerformanceRateMultipleUuidFail() {
+		Node measureReferenceResultsNode = createCorrectMeasureReference(MULTIPLE_POPULATION_DENOM_EXCEPTION_GUID)
+			.replaceSubPopulationPerformanceRate(MULTIPLE_POPULATION_DENOM_EXCEPTION_NUMER1_GUID, "fail1")
+			.build();
+
+		Set<Detail> details = objectUnderTest.validateSingleNode(measureReferenceResultsNode);
+
+		String expectedUuids = StringHelper.join(
+			Lists.newArrayList(MULTIPLE_POPULATION_DENOM_EXCEPTION_NUMER1_GUID, MULTIPLE_POPULATION_DENOM_EXCEPTION_NUMER3_GUID, MULTIPLE_POPULATION_DENOM_EXCEPTION_NUMER2_GUID),
```
Comment (clydet) [inline converter/src/test/java/gov/cms/qpp/conversion/validate/QualityMeasureIdValidatorTest.java]: it's just so long

**Pull request merged**